File: lib/data/mock/mock_boards_repository.dart
import '../../core/roles.dart';
import '../models/bi_board.dart';
import '../repositories/boards_repository.dart';
import '_latency.dart';
import 'mock_data.dart';

class MockBoardsRepository implements BoardsRepository {
  bool simulateError = false;

  @override
  Future<List<BiBoard>> getBoards(UserRole role) async {
    await simulateLatency();
    if (simulateError) {
      throw Exception('Mock: failed to load boards');
    }
    return MockData.boardsForRole(role);
  }

  @override
  Future<BiBoard?> getBoard(String id, UserRole role) async {
    await simulateLatency();
    if (simulateError) {
      throw Exception('Mock: failed to load board');
    }
    final boards = MockData.boardsForRole(role);
    for (final b in boards) {
      if (b.id == id) return b;
    }
    return null;
  }
}

File: Data/Mock/MockBoardsRepository.cs
using OrquestadorFrontend.Core;
using OrquestadorFrontend.Data.Models;
using OrquestadorFrontend.Data.Repositories;

namespace OrquestadorFrontend.Data.Mock;

public sealed class MockBoardsRepository : IBoardsRepository
{
    // Flip to true to demo the ErrorState during the presentation.
    public bool SimulateError { get; set; }

    public async Task<IReadOnlyList<BiBoard>> GetBoardsAsync(UserRole role)
    {
        await MockLatency.SimulateAsync();
        if (SimulateError) throw new InvalidOperationException("Mock: failed to load boards");
        return MockData.BoardsForRole(role);
    }

    public async Task<BiBoard?> GetBoardAsync(string id, UserRole role)
    {
        await MockLatency.SimulateAsync();
        if (SimulateError) throw new InvalidOperationException("Mock: failed to load board");
        return MockData.BoardsForRole(role).FirstOrDefault(b => b.Id == id);
    }
}
